apps/condo/domains/meter/components/CreateMeterReadingsActionBar.tsx

```diff
@@ -41,11 +48,20 @@ export const CreateMeterReadingsActionBar = ({
         >
             {
                 ({ getFieldsValue, getFieldValue, getFieldError }) => {
-                    let isSubmitButtonDisabled, isCreateMeterButtonDisabled, errors, requiredErrorMessage
+                    let isSubmitButtonDisabled, errors, requiredErrorMessage
+
+                    const metersByIds = {}
+                    for (const meter of meters) {
+                        metersByIds[meter.id] = meter
+                    }
+
+                    const hasWrongNumberOfValues = Object.keys(newMeterReadings).reduce((result, meterId) => {
+                        return result || Object.keys(newMeterReadings[meterId]).length !== metersByIds[meterId].numberOfTariffs
```

Comment:
No guarantees in component , that `newMeterReadings` has `meterId` in it, as well as in `metersByIds`

Reply:
`newMeterReadings` contains data from user. All input fields generated from `meters`.

Reply:
If you somehow pass something wrong - it will throw... wrong meterId for example...

It's a bit strange to assume inside here, that all logic above are done rights, espesially when its constantly changing...

Reply:
fixed